.github/workflows/rust.yml

```diff
@@ -21,145 +21,286 @@ on:
   push:
     paths:
       - '.github/workflows/rust.yml'
-      - 'ci/docker/*rust*'
-      - 'ci/scripts/rust_*.sh'
-      - 'ci/scripts/util_*.sh'
       - 'rust/**'
       - 'format/Flight.proto'
   pull_request:
     paths:
       - '.github/workflows/rust.yml'
-      - 'ci/docker/*rust*'
-      - 'ci/scripts/rust_*.sh'
-      - 'ci/scripts/util_*.sh'
       - 'rust/**'
       - 'format/Flight.proto'
 
-env:
-  DOCKER_BUILDKIT: 0
-  COMPOSE_DOCKER_CLI_BUILD: 1
-  ARCHERY_DOCKER_USER: ${{ secrets.DOCKERHUB_USER }}
-  ARCHERY_DOCKER_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
-
 jobs:
 
-  debian:
-    name: AMD64 Debian 10 Rust ${{ matrix.rust }}
+  # build the library, a compilation step used by multiple steps below
+  linux-build-lib:
+    name: Linux ${{ matrix.arch }} Rust ${{ matrix.rust }} build libraries
```

Comment:
I'd prefer the previous title to better align with the rest of the builds, e.g.:
`AMD64 Debian 10 Rust ${{ matrix.rust }} build libraries`

Reply:
Agreed. The issue with hard-coding AMD64 is that it does not depend on `${{ matrix.arch }}`. Would it be ok to be amd64, since that is the name of the image?

Reply:
I see, lowercase is fine then.

Reply:
On the other hand we won't be able to execute non-amd64 build on github actions since we would need hosted agents with different arch, so you can also hardcode 'AMD64'.

Reply:
That is a good point. Reverted that. :)